File: backend/internal/model/match.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Match struct {
	ID           uuid.UUID `json:"id"`
	Player1ID    uuid.UUID `json:"player1_id"`
	Player2ID    uuid.UUID `json:"player2_id"`
	WinnerID     uuid.UUID `json:"winner_id"`
	Player1Score int       `json:"player1_score"`
	Player2Score int       `json:"player2_score"`
	PlayedAt     time.Time `json:"played_at"`
	CreatedAt    time.Time `json:"created_at"`
	CreatedBy    uuid.UUID `json:"created_by"`
}

type MatchWithDetails struct {
	Match
	Player1 User `json:"player1"`
	Player2 User `json:"player2"`
}

File: frontend/src/types/match.ts
import { User } from "./user";

export interface Match {
  id: string;
  player1_id: string;
  player2_id: string;
  winner_id: string;
  player1_score: number;
  player2_score: number;
  played_at: string;
  created_at: string;
  created_by: string;
  player1: User;
  player2: User;
}

export interface CreateMatchRequest {
  player1_id: string;
  player2_id: string;
  player1_score: number;
  player2_score: number;
  played_at: string;
}

export interface HeadToHead {
  player1: User;
  player2: User;
  total_matches: number;
  player1_wins: number;
  player2_wins: number;
  player1_points: number;
  player2_points: number;
  matches: Match[];
}
